boa3/compiler/codegenerator/vmcodemap.py

```diff
@@ -0,0 +1,190 @@
+from typing import List, Dict, Optional
+
+from boa3.neo.vm.VMCode import VMCode
+
+
+class VMCodeMap:
+    def __init__(self):
+        self._vm_code_list: List[VMCode] = []
+        self._vm_code_addresses: List[int] = []
+
+        # optimization so it's not needed to iterate over everything in search of targets
+        self._vm_code_with_target: List[VMCode] = []
+
+    def __len__(self) -> int:
+        return min(len(self._vm_code_list), len(self._vm_code_addresses))
```

Comment:
I don't understand why this `min` is needed. From what I understood, they should be the same length